src/main/java/com/philips/research/bombase/core/nuget/domain/NugetAPI.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2020-2021, Koninklijke Philips N.V., https://www.philips.com
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.philips.research.bombase.core.nuget.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.philips.research.bombase.core.meta.PackageMetadata;
+import com.philips.research.bombase.core.meta.registry.Field;
+import pl.tlinkowski.annotation.basic.NullOr;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public interface NugetAPI {
+    // TODO Is this a proper value?
+    int NUGET_SCORE = 80;
+
+    @GET("registration5-semver1/{project}/{version}.json")
+    Call<CatalogResponseJson> getCatalogEntry(@Path("project") String project,
+                                              @Path("version") String version);
+    @GET("{catalogEntryUrl}")
+    Call<ResponseJson> getDefinition(@Path("catalogEntryUrl") String catalogEntryUrl);
+
+    @SuppressWarnings("NotNullFieldNotInitialized")
```

Comment:
By annotating fields with `@NullOr`, Intellij can infer potential `null` values and raise warnings for missing null checks.

Please add `@NullOr` annotations to the fields that may not always be present (or non-null) in the response.